aya/src/programs/tc.rs

```diff
@@ -228,6 +228,82 @@ define_link_wrapper!(
     TcLinkId
 );
 
+/// # Examples
+///
+/// ```no_run
+/// # #[derive(Debug, thiserror::Error)]
+/// # enum Error {
+/// #     #[error(transparent)]
+/// #     IO(#[from] std::io::Error),
+/// #     #[error(transparent)]
+/// #     Map(#[from] aya::maps::MapError),
+/// #     #[error(transparent)]
+/// #     Program(#[from] aya::programs::ProgramError),
+/// #     #[error(transparent)]
+/// #     Bpf(#[from] aya::BpfError)
+/// # }
+/// # let mut bpf = aya::Bpf::load(&[])?;
+/// # tc::qdisc_add_clsact("eth0")?;
+/// # let prog: &mut SchedClassifier = bpf.program_mut("redirect_ingress").unwrap().try_into()?;
+/// # prog.load()?;
+///
+/// use aya::programs::tc::{SchedClassifierLink};
+/// use aya::programs::{tc, Link, SchedClassifier, TcAttachType};
+///
+/// // SchedClassifier::attach returns a SchedClassifierLinkId that can be used to
+/// // detach the program.
+/// let tc_link_id = prog.attach("eth0", TcAttachType::Ingress)?;
+///
+/// // The user may take ownership of the lifetime of a link using
+/// // SchedClassifier::take_link, which returns a SchedClassifierLink
+/// let tc_link = prog.take_link(tc_link_id)?;
+///
+/// // Once ownership is taken, SchedClassifierLink::detach can be used to detach the
+/// // link. If needed, the link can be reconstructed via the SchedClassifierLink::new
+/// // using the if_name, attach_type, priority, and handle. The user knows the first
+/// // two because they were required to execute attach(); however, the user may not
+/// // know the priority, and/or handle if they let the system choose them as happens
+/// // with SchedClassifier::attach. If needed, These items may be retrieved as follows:
+/// let priority = tc_link.priority();
+/// let handle = tc_link.handle();
+///
+/// // Then, the link can be reconstructed as follows:
+/// let tc_link = SchedClassifierLink::new("eth0", TcAttachType::Ingress, priority, handle)?;
+///
+/// // And, the user can then detatch the link as follows:
+/// tc_link.detach()?;
+///
+/// # Ok::<(), Error>(())
+/// ```
+impl SchedClassifierLink {
+    /// Creates a new `SchedClassifierLink` instance
```

Comment:
```suggestion
    /// Creates a new link from the provided values.
```

You could elaborate more in the docs here.

- Explain the use case - This API is used to construct a [`SchedClassifierLink`] where the `if_name`, `attach_type`, `priority` and `handle` are already known. This may have been found from a link created by [`SchedClassifier::attach`], the output of the `tc filter` command or from the output of another BPF loader.
- Explain what happens if any of the values are incorrect/invalid. Specifically I'd add a *WARNING* if you pass a handle/priority in use by another program that you do not own, bad things may happen.